src/main.rs

```diff
@@ -56,10 +56,13 @@ async fn write_state(config: &CrunchyConfiguration) {
 
     geo_cache.configure_providers(&config.geoip_config);
 
+    const NUM_THREADS: usize = 8;
```

Comment:
I wonder if there is any sense to have different value for that in different modules.
Because if not, maybe it could be better to put it in some constants.rs file and use it from there. It could also simplify API because we'd not pass it as an argument.

Reply:
I like that idea a lot. Will do.
And, tbh, I've been thinking about the threading, and don't really like 'polluting' the spectre api with having to specify the thread count.  The teexgraph library I used just queried  the number of CPUs via the omp (C++) library:
```
   const int cpus = omp_get_num_procs();
```
I don't exactly know what the Rust equivalents might be, but I did see this:
https://docs.rs/num_cpus/latest/num_cpus/

So, in any event, I think the end game would be to remove `num_threads` throughout, and the compute call **just does it**.  Something we can discuss.